olp-cpp-sdk-core/src/network2/Network.cpp

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/core/network2/Network.h>
+
+namespace olp {
+namespace network2 {
+
+Network::~Network() = default;
```

Comment:
Just curious - why not to move it to Network.h?

Reply:
that will make destructor definition part of API

Reply:
Is this bad?
I dont see any notion of this beeing bad, actually all abstract interfaces are mostly header only.
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Ri-abstract